packages/cloudscape-react-ts-website/samples/src/Auth.tsx

```diff
@@ -69,58 +35,35 @@ const Auth: React.FC<any> = ({ children }) => {
             runtimeCtx.userPoolId &&
             runtimeCtx.userPoolWebClientId &&
             runtimeCtx.identityPoolId) {
-          Amplify.configure({
-            Auth: {
-              region: runtimeCtx.region,
-              userPoolId: runtimeCtx.userPoolId,
-              userPoolWebClientId: runtimeCtx.userPoolWebClientId,
-              identityPoolId: runtimeCtx.identityPoolId,
-            },
-          });
-          AmplifyAuth.currentUserInfo()
-            .then(user => setRuntimeContext({ ...runtimeCtx, user }))
-            .catch(e => console.error(e));
+          setRuntimeContext(runtimeCtx as RuntimeContext);
         } else {
           console.warn('runtime-config.json should have region, userPoolId, userPoolWebClientId & identityPoolId.');
         }
       })
       .catch(() => {
         console.log('No runtime-config.json detected');
-        setRuntimeContext({});
+        setRuntimeContext(undefined);
       });
   }, [setRuntimeContext]);
 
-  useEffect(() => {
-    Hub.listen('auth', (data) => {
-      switch (data.payload.event) {
-        case 'signIn':
-          AmplifyAuth.currentUserInfo()
-            .then(user => setRuntimeContext((prevRuntimeContext: any) => ({ ...prevRuntimeContext, user })))
-            .catch(e => console.error(e));
-          break;
-        case 'signOut':
-          window.location.reload();
-          break;
-      }
-    });
-  }, []);
-
   const AuthWrapper: React.FC<any> = useCallback(({ children: _children }) => runtimeContext?.userPoolId ?
-    <ThemeProvider theme={theme}>
-      <Authenticator variation="modal" hideSignUp>
-        {_children}
-      </Authenticator>
-    </ThemeProvider> :
+    <CognitoAuth
+      header={Config.applicationName}
+      userPoolId={runtimeContext.userPoolId}
+      clientId={runtimeContext.userPoolWebClientId}
+    >
+      {_children}
+    </CognitoAuth> :
     <>
       {
         runtimeContext ?
```

Comment:
This can be moved outside of CognitoAuth given that if runtimeContext is null, the CognitoAuth should not be rendered at all.

Reply:
Not sure what you mean by this. Can you please elaborate?